yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -975,6 +975,10 @@ private[spark] class Client(
           case e: ApplicationNotFoundException =>
             logError(s"Application $appId not found.")
             return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
+          case e: Exception if (e.isInstanceOf[InterruptedException]
+            || e.getCause.isInstanceOf[InterruptedException]) =>
+            logInfo("The reporter thread is interrupted, we assume app is finished.")
+            return (YarnApplicationState.FINISHED, FinalApplicationStatus.SUCCEEDED)
```

Comment:
how about we change the status to be UNDEFINED.  

Since we really don't know the status it seems like returning the undefined in this case make more sense. Hopefully the user would then go look more at the RM or spark job details.